src/loaders/amf_load.c

```diff
@@ -314,10 +344,21 @@ static int amf_load(struct module_data *m, HIO_HANDLE *f, const int start)
 		if (libxmp_alloc_track(mod, i, 64) < 0)	/* FIXME! */
 			return -1;
 
-		size = hio_read24l(f);
+		/* Previous versions loaded this as a 24-bit value, but it's
+		 * just a word. The purpose of the third byte is unknown, and
+		 * DSMI just ignores it.
+		 */
+		size = hio_read16l(f);
+		hio_read8(f);
+
 		if (hio_error(f))
 			return -1;
 
+		/* Version 0.1 AMFs apparently have an extra event when the
```

Comment:
Yeah, I agree that would definitely be a better way to explain it, and it makes sense. I'll patch this comment after I've found a copy of M2AMF that produces 0.1 AMFs on this BBS CD archive (or more likely, after I give up, because it feels like I've been finding every version except for the ones relevant to AMF 0.1 and 1.0).

Reply:
I updated this comment.